File: backend/src/main/java/com/zennyt/engagement/domain/model/Comment.java
package com.zennyt.engagement.domain.model;

import java.time.Instant;
import java.util.Objects;
import java.util.UUID;

public record Comment(UUID id, UUID postId, UUID authorId, String content, Instant createdAt) {
    public Comment {
        Objects.requireNonNull(id, "id");
        Objects.requireNonNull(postId, "postId");
        Objects.requireNonNull(authorId, "authorId");
        if (content == null || content.isBlank()) throw new IllegalArgumentException("Commentaire obligatoire");
        Objects.requireNonNull(createdAt, "createdAt");
    }

    public static Comment create(UUID postId, UUID authorId, String content) {
        return new Comment(UUID.randomUUID(), postId, authorId, content.trim(), Instant.now());
    }
}

File: mobile/lib/features/home/domain/entities/comment.dart
import 'package:equatable/equatable.dart';

/// Entité métier Commentaire.
class Comment extends Equatable {
  final String id;
  final String postId;
  final String authorId;
  final String authorName;
  final String? authorAvatarUrl;
  final String content;
  final DateTime createdAt;

  const Comment({
    required this.id,
    required this.postId,
    required this.authorId,
    required this.authorName,
    this.authorAvatarUrl,
    required this.content,
    required this.createdAt,
  });

  @override
  List<Object?> get props => [
        id,
        postId,
        authorId,
        authorName,
        authorAvatarUrl,
        content,
        createdAt,
      ];
}
